src/containers/modals/deposit/steps/ApproveStep.js

```diff
@@ -0,0 +1,181 @@
+    }
+  }
+
+  async function doReset () {
+    const res = await dispatch(resetApprove(weiAmount, currency, gasPrice));
+    if (res) {
+      dispatch(openAlert('ERC20 approval reset successful.'));
+      checkAllowance();
+    }
+  }
+
+  async function doDeposit () {
+    const res = await dispatch(depositErc20(weiAmount, currency, gasPrice));
+    if (res) {
+      dispatch(openAlert('ERC20 deposit submitted. Check the Deposits tab to see the status of your deposit.'));
+      handleClose();
+    }
+  }
+
+  const tokenName = tokenInfo.name || tokenInfo.currency;
+
+  const renderCancelButton = (
+    <Button
+      onClick={handleClose}
+      type='outline'
+      style={{ flex: 0 }}
+    >
+      CANCEL
+    </Button>
+  );
+
+  const renderGasPicker = (
+    <GasPicker
+      selectedSpeed={selectedSpeed}
+      setSelectedSpeed={setSelectedSpeed}
+      setGasPrice={setGasPrice}
+    />
+  );
+
+  return (
+    <>
+      <h2>Approval</h2>
+
+      {!allowance && (
+        <div>Checking allowance...</div>
+      )}
+
+      {allowance === '0' && (
+        <>
+          <p>
+            {`In order to deposit ${value.toString()} ${tokenName}, you first need to approve this amount. Click below to submit an approval transaction.`}
+          </p>
+          {renderGasPicker}
+          <div className={styles.buttons}>
+            {renderCancelButton}
+            <Button
+              onClick={doApprove}
+              type='primary'
+              style={{ flex: 0 }}
+              tooltip='Your approval transaction is still pending. Please wait for confirmation.'
+              loading={approveLoading}
+              disabled={approveLoading}
+            >
+              APPROVE
+            </Button>
+          </div>
+        </>
+      )}
+
+      {allowance && allowance !== '0' && allowance !== weiAmount.toString() && (
+        <>
+          <p>
+            {`You are already approved to deposit ${logAmount(allowance, tokenInfo.decimals)} ${tokenName}. Since this amount does not match how much you want to deposit now, you will need to reset your allowance.`}
+          </p>
+          <p>
+            You will be prompted with 2 approval transactions. One to reset the allowance to 0, and another for the new amount.
+          </p>
+          {renderGasPicker}
+          <div className={styles.buttons}>
+            {renderCancelButton}
+            <Button
+              onClick={doReset}
+              type='primary'
+              style={{ flex: 0 }}
+              tooltip='Your reset transaction is still pending. Please wait for confirmation.'
+              loading={resetLoading}
+              disabled={resetLoading}
+            >
+              RESET
+            </Button>
+          </div>
+        </>
+      )}
+
+      {allowDeposit && (
+        <>
+          <p>
+            {`You are approved to deposit ${logAmount(allowance, tokenInfo.decimals)} ${tokenName}. Click below to make the deposit.`}
```

Comment:
`You are approved to deposit ${logAmount(allowance, tokenInfo.decimals)} ${tokenName}. Click below to make the deposit.` -> 
`Your approval request for ${logAmount(allowance, tokenInfo.decimals)} ${tokenName} was confirmed. Click below to make the deposit.`

Simply changing this to avoid using `deposit` twice in two sentences in a row.